docs/pages/CodeAnalysis.md

```diff
@@ -0,0 +1,226 @@
+
+- For each entity set in the `entitySets` object, each sub-entry is checked for a `default.template` property,
+- if one is found, the containing string is interpreted as the short name of a template component in the `sap.fe.templates` library,
+- a dependency to that template component is added to the analyzed app.
+
+For a full analysis, "routing" also should be taken into account. Only when a sub-entry of the entity set
+is referenced by a route, the template for that entry will be used. Routes thereby could form entry points.
+
+```json
+{
+    "sap.fe" : {
+        "entitySets" : {
+            "C_AIVS_MDBU_ArtistTP" : {
+                "feed": {
+                    "default": {
+                        "template": "ListReport"
+                    }
+                },
+                "entry" : {
+                    "default" : {
+                        "outbound" : "musicV2Display"
+                    }
+                }
+            }
+        },
+        "routing" : {
+            "routes" :{
+                "ArtistList": {
+                    "target": "C_AIVS_MDBU_ArtistTP/feed"
+                }
+            }
+        }
+    }
+}
+```
+
+The template component is analyzed in the following way:
+
+- precondition: the template component class is defined in an AMD-style module, using define or sap.ui.define
+- precondition: the module 'sap/fe/core/TemplateAssembler' is imported
+- precondition: a call to TemplateAssembler.getTemplateComponent is used to define the component class
+- precondition: that call is used in a top-level return statement of the factory function
+- precondition: necessary parameters to that call are given as an object literal (no further coding)
+- precondition: the settings define a managed property 'metadata.properties.templateName' with a
+                defaultValue of type string
+The default value of the property represents the template view of the template component.
+The manifest of the template app could in theory specify an alternative template as setting.templateName,
+but as of June 2017, this possibility is currently not used.
+
+### Smart Template Analyzer
+
+The [Smart Template Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/SmartTemplateAnalyzer.js) analyzes a Smart Template app and its underlying template components to collect dependency information. It searches for a `manifest.json` located in the same folder as the `Component.js`. If it is found and if it is a valid JSON, an "sap.ui.generic.app" section is searched and evaluated in the following way
+
+- For each page configuration, the configured component is added as a dependency to the template app module,
+- if the page configuration contains a `templateName`, a dependency to that template view is added to the app,
+- otherwise, the class definition of the component is analyzed to find a default template view name. if found, a dependency to that view is added to the app module.
```

Comment:
```suggestion
- For each page configuration, the configured component is added as a dependency to the template app module.
- If the page configuration contains a `templateName`, a dependency to that template view is added to the app.
- Otherwise, the class definition of the component is analyzed to find a default template view name. if found, a dependency to that view is added to the app module.
```